app/views/track.js

```diff
@@ -118,6 +127,45 @@ const TrackView = Mn.LayoutView.extend({
         e.stopPropagation();
     },
 
+    setClass() {
+        this.className = ''
+        this.order = [];
+
+        // Check if track is in upNext
+        let tracks = application.upNext.get('tracks');
+        tracks.each((track, index) => {
+            if (track.get('_id') == this.model.get('_id')) {
+                this.className += ' playlist-upNext';
+                this.order.push({ id: 'upNext', order: index });
+                return;
```

Comment:
i think your code is exaclty the same as mine :p 
I'm going to test with findwhere.